### DISCUSSION TITLE
Can we decouple Statistics from FDA main project

### DISCUSSION THREAD

**Discussion by HenryGeorgist** [HydrologicEngineeringCenter/HEC-FDA#95] [Q&A] [answered]
I recommend that we put the statistics project into a separate repository. The reason I am starting the conversation is that I want to reference the statistics library in the new fda-model project. If someone can help me do that without moving the statistics library I am also open to that solution

**Comment by Brennan1994** [comment]
We could make it a NuGet Package and host it in this Repository, or we could make a repository explicitly for that source code and host the NuGet Package there. 

Having it's own repo sounds cleaner to me, but there is a benefit to having it in the same solution, able to easily reference it's source code. I can probably make either alternative happen, I'll look to you guys who have been using the code more to weigh in on the value of keeping it all together.

**Reply by Brennan1994** [reply]
If we host it here, that does not preclude us from hosting another, different,  NuGet package here later. Just FYI

**Reply by HenryGeorgist** [reply]
I am totally open to keeping it part of the project as long as I can safely reference it externally. @rnugent3 can weigh in when we have had a chance to discuss as a team the pros and cons.

**Reply by rnugent3** [reply]
Sounds like it was decided to keep the statistics project in the HEC-FDA repository and fix the ability to reference the library from fda-model repository?

**Reply by Brennan1994** [reply]
Long term decision hasn't really been made. I hosted a NuGet Package on Nexus that we now reference from the model. There's no infrastructure there though, I just did it manually until we know the ultimate fate of the statistics project. Seems to me there's a chance we're better off poaching what we like and dumping the rest. I'll create a discussion with alternatives and a recommendation when we need to make that decision. I think we can put it off for the time being.

**Comment by Brennan1994** [comment]
The only messy bit about moving Statistics out of  this repo cleanly is the Utilities reference. I propose we break that out into it's own repo as well.

**Reply by rnugent3** [reply]
I'll support whatever you think is best for us, @Brennan1994.

**Comment by Brennan1994** [comment, marked as answer]
To keep this discussion up to date: I've broken out Utilities and Statistics into their own  Repos. As currently designed, a new package will be generated and hosted in the repo every time a commit is pushed to main with a new version number. Projects referencing the package will need to add the HydrologicEngineeringCenter packages repo to their NuGet sources, and verify their credentials (use a Github personal access token) 

To update a projects references to a new version in Visual Studio, go to tools| manage packages for solution | Updates. If you have the github page selected as your source in the top right hand corner of the GUI you should see all available packages form HEC's Github (all ours atm) and change your reference however you like.

**Comment by rnugent3** [comment]
Sounds like you’ve automated the workflow? 

Thanks Brennan!!!

**Comment by Brennan1994** [comment]
Another update. I took some inspiration from the RAS team's tooling today and automated the versioning of packages. Now every push to main will publish a package with the version number in the project file plus -alpha-DATETIME. This means the latest is always immediately available, but you will still have to update your project to use it manually. I personally like having to do the update manually, and I think it's good to make sure we're always aware when our dependency is changing. 

Packages now publish automatically with every push to main.
